scrapy/utils/retry.py

```diff
@@ -0,0 +1,54 @@
+from scrapy.utils.python import global_object_name
+
+
+class RetryHandler(object):
+    """utilities to handle retries"""
+
+    def __init__(self, spider, request):
```

Comment:
For me, it seems that the most important dependency is the request.
So, I would change the precedence to put the request first.